resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,235 @@
+    #     "West Bank Standard Time", "South Africa Standard Time", "FLE Standard Time", "Israel Standard Time", "Kaliningrad Standard Time", 
+    #     "Sudan Standard Time", "Libya Standard Time", "Namibia Standard Time", "Arabic Standard Time", "Turkey Standard Time", 
+    #     "Arab Standard Time", "Belarus Standard Time", "Russian Standard Time", "E. Africa Standard Time", "Iran Standard Time", 
+    #     "Arabian Standard Time", "Astrakhan Standard Time", "Azerbaijan Standard Time", "Russia Time Zone 3", "Mauritius Standard Time", 
+    #     "Saratov Standard Time", "Georgian Standard Time", "Caucasus Standard Time", "Afghanistan Standard Time", "West Asia Standard Time", 
+    #     "Ekaterinburg Standard Time", "Pakistan Standard Time", "India Standard Time", "Sri Lanka Standard Time", "Nepal Standard Time", 
+    #     "Central Asia Standard Time", "Bangladesh Standard Time", "Omsk Standard Time", "Myanmar Standard Time", "SE Asia Standard Time", 
+    #     "Altai Standard Time", "W. Mongolia Standard Time", "North Asia Standard Time", "N. Central Asia Standard Time", 
+    #     "Tomsk Standard Time", "China Standard Time", "North Asia East Standard Time", "Singapore Standard Time", "W. Australia Standard Time", 
+    #     "Taipei Standard Time", "Ulaanbaatar Standard Time", "North Korea Standard Time", "Aus Central W. Standard Time", 
+    #     "Transbaikal Standard Time", "Tokyo Standard Time", "Korea Standard Time", "Yakutsk Standard Time", "Cen. Australia Standard Time", 
+    #     "AUS Central Standard Time", "E. Australia Standard Time", "AUS Eastern Standard Time", "West Pacific Standard Time", 
+    #     "Tasmania Standard Time", "Vladivostok Standard Time", "Lord Howe Standard Time", "Bougainville Standard Time", 
+    #     "Russia Time Zone 10", "Magadan Standard Time", "Norfolk Standard Time", "Sakhalin Standard Time", "Central Pacific Standard Time", 
+    #     "Russia Time Zone 11", "New Zealand Standard Time", "UTC+12", "Fiji Standard Time", "Kamchatka Standard Time", 
+    #     "Chatham Islands Standard Time", "UTC+13", "Tonga Standard Time", "Samoa Standard Time", "Line Islands Standard Time"
+    # )]
+    [string] $TimeZone = (Get-TimeZone).Id
+
+    [DscProperty()]
+    [TimeZoneAutomatically] $SetTimeZoneAutomatically = [TimeZoneAutomatically]::NTP
+
+    [DscProperty()]
+    [nullable[bool]] $ShowSystemTrayDateTime
+
+    static hidden [string] $SetTimeZoneAutomaticallyProperty = 'Type'
+    static hidden [string] $ShowSystemTrayDateTimeProperty = 'ShowSystrayDateTimeValueName'
+
+    [Time] Get()
+    {
+        $currentState = [Time]::New()
+        $currentState.SetTimeZoneAutomatically = [Time]::GetTimeZoneAutoUpdateStatus()
+        $currentState.TimeZone = (Get-TimeZone).Id
+        $currentState.ShowSystemTrayDateTime = [Time]::GetShowSystemTrayDateTimeStatus()
+
+        return $currentState
+    }
+
+    [void] Set()
+    {
+        if ($this.Test())
```

Comment:
I appreciate attempting to return early, but doesn’t this result in two calls to `$this.get()`?

Reply:
It does indeed. I tried figuring out a way to get the current state early on without having multiple calls. I don't want to compare apples with apples, but the same behaviour was happening in `Microsoft.Windows.Setting.Accessibility`:

```powershell
if ($this.Test())
{
    return
}

if (-not (Test-Path -Path $global:MagnifierRegistryPath))
{
    New-Item -Path $global:MagnifierRegistryPath -Force | Out-Null
}

if ($this.Magnification -ne [MagnificationValue]::KeepCurrentValue)
{
    $desiredMagnification = switch ([MagnificationValue]($this.Magnification))
    {
        None { 0 }
        Low { 100 }
        Medium { 200 }
        High { 300 }
    }

    Set-ItemProperty -Path $global:MagnifierRegistryPath -Name $this.MagnificationProperty -Value $desiredMagnification -Type DWORD
}

$currentState = $this.Get()
```

If you have a better way, please shoot Kaleb :)

Reply:
No worries, was just curious